src/extensions/surveys.tsx

```diff
@@ -219,3 +118,752 @@ export function generateSurveys(posthog: PostHog) {
         callSurveys(posthog, false)
     }, 3000)
 }
+
+const defaultBackgroundColor = '#eeeded'
+
+export function QuestionHeader({
+    question,
+    description,
+    backgroundColor,
+}: {
+    question: string
+    description?: string | null
+    backgroundColor?: string
+}) {
+    return (
+        <div style={{ backgroundColor: backgroundColor || defaultBackgroundColor }}>
+            <div className="survey-question">{question}</div>
+            {description && <div className="description" dangerouslySetInnerHTML={{ __html: description }} />}
+        </div>
+    )
+}
+
+export function Cancel({ onClick }: { onClick: () => void }) {
+    return (
+        <div className="cancel-btn-wrapper">
+            <button className="form-cancel" onClick={onClick}>
+                {cancelSVG}
+            </button>
+        </div>
+    )
+}
+
+export function BottomSection({
+    text,
+    submitDisabled,
+    appearance,
+    onSubmit,
+    link,
+}: {
+    text: string
+    submitDisabled: boolean
+    appearance: SurveyAppearance
+    onSubmit: () => void
+    link?: string | null
+}) {
+    const backgroundColorRef = Preact.createRef()
+    const [textColor, setTextColor] = useState('white')
+
+    useEffect(() => {
+        const color = getTextColor(backgroundColorRef.current)
+        setTextColor(color)
+    }, [appearance])
```

Comment:
This whole block was pretty confusing 😅 

I sort of get what it does I think - using the element to find the color and then setting an appropriate text color.

Feels like this would be a strong use case for `useContext` passing the appearance object through so that you can determine it without having to query the DOM inefficiently? 

For now I guess it's fine but could be a good thing to follow up on as this could be an easy source of rendering bugs in the future.

Reply:
added as a TODO